streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -142,7 +143,12 @@ public void openDB(ProcessorContext context) {
         fOptions.setWaitForFlush(true);
 
         final Map<String, Object> configs = context.appConfigs();
-        final Class<RocksDBConfigSetter> configSetterClass = (Class<RocksDBConfigSetter>) configs.get(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG);
+        final Object configSetterValue = configs.get(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG);
+        final Class<RocksDBConfigSetter> configSetterClass = (Class<RocksDBConfigSetter>) ConfigDef.parseType(
```

Comment:
I don't agree with this comment.

The problem (from my point of view) is a follows:

We have a parameter of type "CLASS". When we call `context.appConfigs().get(...)` for this parameter we should expect a Class type back. This should be independent of the
original type that was used by the user to set the parameter. If
`StreamsConfig` accepts a String for `Type.CLASS` it is `StreamsConfig`'s
responsibility to convert the type -- but not the caller's responsibility IMHO.

The underlying issues is simply, that we return the unparsed `original`
value here, instead of a "parsed parameter map".

So why do you think it's a good idea that RockDBStore is responsible for
this?

Discussed this with @guozhangwang offline -- he agrees with my argument.

@twbecker We can do a follow up on this with a new JIRA. Would you like to work on this? If yes, please feel free to create a JIRA and assign it to yourself. Otherwise, I can create the Jira, too.